src/utils/EventUtils.ts

```diff
@@ -55,11 +57,25 @@ export async function paginatedEventQuery(contract: Contract, filter: EventFilte
   }
 }
 
-export interface EventSearchConfig {
-  fromBlock: number;
-  toBlock: number | null;
-  maxBlockLookBack: number;
-  concurrency?: number | null;
+export function getPaginatedBlockRanges(searchConfig: EventSearchConfig): number[][] {
+  const nextSearchConfig = { fromBlock: searchConfig.fromBlock, toBlock: searchConfig.toBlock };
+
+  if (searchConfig.maxBlockLookBack !== undefined)
+    nextSearchConfig.toBlock = Math.min(searchConfig.toBlock, searchConfig.fromBlock + searchConfig.maxBlockLookBack);
+
+  if (searchConfig.maxBlockLookBack === 0) throw new Error("Cannot set maxBlockLookBack = 0");
+
+  const returnValue: number[][] = [];
+  while (nextSearchConfig.fromBlock <= searchConfig.toBlock) {
+    returnValue.push([nextSearchConfig.fromBlock, nextSearchConfig.toBlock]);
+    nextSearchConfig.fromBlock = nextSearchConfig.toBlock + 1;
+    if (searchConfig.maxBlockLookBack !== null)
+      nextSearchConfig.toBlock = Math.min(
+        searchConfig.toBlock,
+        nextSearchConfig.toBlock + 1 + searchConfig.maxBlockLookBack
```

Comment:
This accomplishes the same thing, but I think it's a little clearer if you just take the nextSearchConfig.fromBlock that you computed on line 71 and add maxBlockLookBack to it. It took me a minute to parse out what this was trying to do, and it's less obviously correct than just making it exactly 1 lookback past the fromBlock that you're about to use.

```suggestion
        nextSearchConfig.fromBlock + searchConfig.maxBlockLookBack
```